mesonbuild/compilers/cpp.py

```diff
@@ -311,6 +311,9 @@ def get_option_link_args(self, target: 'BuildTarget', env: 'Environment', subpro
             return libs
         return []
 
+    def is_libcpp_enable_assertions_deprecated(self) -> bool:
+        return version_compare(self.version, ">=18")
```

Comment:
This checks clang version and not libc++ version, right? They might mismatch so this check is technically incorrect, especially on systems with more than one version of clang installed.

(This is the previous behavior so it's probably fine to keep it here, just noting)

Reply:
Yeah, I noticed that too. In principle, we would want the check to be: "if libc++ >= 18 is used, or Apple's libc++ >=17 is used,  use `_LIBCPP_HARDENING_MODE` instead of `_LIBCPP_ENABLE_ASSERTIONS`".

Reply:
That merits a comment, but it's already handled in my pending refactoring of it.